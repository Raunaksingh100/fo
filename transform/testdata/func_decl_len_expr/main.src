package main

func A[T] (x []T) int {
	return len(x)
}

func B[T] (x [3]T) int {
	return len(x)
}

func C[T, U] (x map[T]U) int {
	return len(x)
}

func main() {}
