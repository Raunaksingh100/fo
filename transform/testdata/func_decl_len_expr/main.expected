package main

import "reflect"

func A(T interface{}) func(x interface{}) int {
	return func(x interface{}) int {
		return reflect.ValueOf(x).Len()
	}
}

func B(T interface{}) func(x interface{}) int {
	return func(x interface{}) int {
		return reflect.ValueOf(x).Len()
	}
}

func C(T interface{}, U interface{}) func(x interface{}) int {
	return func(x interface{}) int {
		return reflect.ValueOf(x).Len()
	}
}

func main() {}
